src/Draco.Examples.Tests/ExamplesTests.cs

```diff
@@ -0,0 +1,76 @@
+using System.Diagnostics;
+using System.Text;
+using DiffEngine;
+
+namespace Draco.Examples.Tests;
+
+public sealed class ExamplesTests
+{
+    public static IEnumerable<object[]> TestData
+    {
+        get
+        {
+            // Get all example projects
+            // We exclude the "Toolchain" folder, in case the user has installed it in the examples directory
+            var exampleProjectDirectories = Directory
+                .GetDirectories("examples", "*", SearchOption.TopDirectoryOnly)
+                .Where(d => Path.GetFileName(d) != "Toolchain");
+            // Each directory contains a projectfile, and a verification file, return each pair
+            return exampleProjectDirectories
+                .Select(directory => new object[]
+                {
+                    // Search for the dracoproj file
+                    Directory.GetFiles(directory, "*.dracoproj").Single(),
+                    // The verification file is always named "verify.txt"
+                    Path.Combine(directory, "verify.txt"),
+                });
+        }
+    }
+
+    public ExamplesTests()
+    {
+        DiffTools.UseOrder(DiffTool.VisualStudioCode, DiffTool.VisualStudio, DiffTool.Rider);
+    }
+
+    [Theory]
+    [MemberData(nameof(TestData))]
+    public async Task RunExample(string projectFile, string verifiedFile)
+    {
+        // Invoke 'dotnet run' on the project
+        var process = new Process
+        {
+            StartInfo = new ProcessStartInfo
+            {
+                FileName = "dotnet",
+                ArgumentList = { "run", "--project", projectFile },
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                UseShellExecute = false,
+                CreateNoWindow = true,
+            },
+        };
+        process.Start();
+
+        var standardOutput = new StringBuilder();
+        while (!process.StandardOutput.EndOfStream)
+        {
+            var line = process.StandardOutput.ReadLine();
+            standardOutput.AppendLine(line);
+        }
+        var gotOutput = standardOutput.ToString();
+
+        // Wait for the process to exit
+        process.WaitForExit();
```

Comment:
Will hit https://github.com/dotnet/runtime/issues/51277 as soon as our example spawn a subprocess.  
Unlikely but I love reminding this bug exists until it's fixed.
There isn't a real workaround to do either